cmd/mount.go

```diff
@@ -431,28 +424,49 @@ var mountCmd = &cobra.Command{
 				log.Err("mount : failed to daemonize application [%v]", err)
 				return Destroy(fmt.Sprintf("failed to daemonize application [%s]", err.Error()))
 			}
+
 			log.Debug("mount: foreground disabled, child = %v", daemon.WasReborn())
-			if child == nil {
+			if child == nil { // execute in child only
 				defer dmnCtx.Release() // nolint
 				setGOConfig()
 				go startDynamicProfiler()
 
 				err = runPipeline(pipeline, ctx)
 				if err != nil {
+					pid := os.Getpid()
+					fname := fmt.Sprintf("/tmp/blobfuse2.%v", pid)
+					if err = ioutil.WriteFile(fname, []byte(err.Error()), 0777); err != nil {
+						log.Err("mount: failed to save mount failure logs [%s]", err.Error())
+					}
+
 					return err
 				}
-			} else {
+
+			} else { // execute in parent only
 				select {
 				case <-sigusr2:
-					log.Info("libfuse mount at %s succeed.", options.MountPath)
+					log.Info("mount: Child [%v] mounted successfully at %s", child.Pid, options.MountPath)
+
 				case <-sigchild:
-					buf := make([]byte, 1024)
-					pipeR.Read(buf)
-					return Destroy(fmt.Sprintf("libfuse mount failed: %s", buf))
+					// Get error string from the child
+					log.Info("mount: Child [%v] terminated from %s", child.Pid, options.MountPath)
+
+					fname := fmt.Sprintf("/tmp/blobfuse2.%v", child.Pid)
+
+					buff, err := ioutil.ReadFile(fname)
+					if err != nil {
+						log.Err("mount: failed to read child [%v] failure logs [%s]", child.Pid, err.Error())
+						return Destroy(fmt.Sprintf("failed to mount, please check logs [%s]", err.Error()))
```

Comment:
I think print out buff is better, what if the temp log file is cleaned up by system? We can keep the log file for more details.

Reply:
this is only when we fail to read the temp file, otherwise we are printing the contents of the temp file